src/relentless/simulate/lammps.py

```diff
@@ -312,6 +381,20 @@ def _convert_to_data_file(self, sim):
                 snap, type_map = lammpsio.Snapshot.from_hoomd_gsd(frame)
                 type_map = {v: k for k, v in type_map.items()}
 
+                # check atom style is compatible with topology data if present
+                if (
+                    snap.has_bonds()
+                    or snap.has_angles()
+                    or snap.has_dihedrals()
+                    or snap.has_impropers()
+                ):
+                    if sim["engine"]["atom_style"] == "atomic":
+                        raise ValueError(
+                            "Atomic atom style is not compatible with topology data."
+                        )
+                # store topology data
+                if snap.has_bonds():
+                    sim[self]["bonds"] = snap.bonds
```

Comment:
Make sure `sim[self]["bonds"]` is either assigned to `None` or broadcast on other ranks. Right now it will be undefined. You may also want to make it private as `_bonds` if we don't want to guarantee it will be available as part of the interface forever.